frontend/appflowy_flutter/lib/mobile/presentation/home/mobile_home_page.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy_backend/dispatch/dispatch.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder2/workspace.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-user/protobuf.dart';
+import 'package:flutter/material.dart';
+
+// TODO(yijing): This is just a placeholder for now.
+class MobileHomeScreen extends StatelessWidget {
+  const MobileHomeScreen({super.key});
+
+  static const routeName = "/MobileHomeScreen";
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: Future.wait([
+        FolderEventGetCurrentWorkspace().send(),
+        getIt<AuthService>().getUser(),
+      ]),
+      builder: (context, snapshots) {
+        if (!snapshots.hasData) {
+          return const Center(child: CircularProgressIndicator());
+        }
+
+        WorkspaceSettingPB? workspaceSetting;
+        UserProfilePB? userProfile;
+        snapshots.data?[0].fold(
+          (workspaceSettingPB) {
+            workspaceSetting = workspaceSettingPB as WorkspaceSettingPB?;
+          },
+          (error) => null,
+        );
+        snapshots.data?[1].fold((error) => null, (userProfilePB) {
+          userProfile = userProfilePB as UserProfilePB?;
+        });
```

Comment:
use fold the simplify the logic.

```suggestion
        final workspaceSetting = snapshots.data?[0].fold<WorkspaceSettingPB?>(
          (workspaceSettingPB) => workspaceSettingPB as WorkspaceSettingPB?,
          (_) => null,
        );
        final userProfile = snapshots.data?[1].fold<UserProfilePB?>(
          (userProfilePB) => userProfilePB as UserProfilePB?,
          (_) => null,
        );
```